packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -169,8 +172,15 @@ function useBlockEditorSettings( settings, postType, postId, renderingMode ) {
 				hiddenBlockTypes: get( 'core', 'hiddenBlockTypes' ),
 				isDistractionFree: get( 'core', 'distractionFree' ),
 				keepCaretInsideBlock: get( 'core', 'keepCaretInsideBlock' ),
-				hasUploadPermissions: canUser( 'create', 'media' ) ?? true,
-				userCanCreatePages: canUser( 'create', 'pages' ),
+				hasUploadPermissions:
+					canUser( 'create', {
+						kind: 'root',
+						name: 'media',
```

Comment:
I've always found it odd that the media entity is a separate root entity, and we don't use the `attachment` post type. Somehow, this special case is more visible now, which I find potentially confusing. Not much to do obviously, since this is about inherited complexity.

Reply:
100%, but WP always treated Media as a separate entity, even though it's just a post type.

P.S. Jorge discovered another odd case recently: https://github.com/WordPress/gutenberg/issues/63438.